pl_spv_forecasts/src/services/meteo_data/visualcrossing-api/fetch_data.py

```diff
@@ -0,0 +1,49 @@
+from pandas import DataFrame, Series, concat, to_datetime
+from datetime import date
+import requests
+from dataclasses import dataclass
+from utils import columns as selected_columns
+
+
+@dataclass
+class VisualCrossingData:
+    """
+    A class to retrieve forecast weather data from the VisualCrossing Weather API.
+    """
+
+    base_url = 'https://weather.visualcrossing.com/VisualCrossingWebServices/rest/services/timeline'
+
+    def call_visualcrossing_api(self, lat: float, lon: float, api_key: str,
+                                start_date: date = None, end_date: date = None) -> requests.Response:
+        if start_date is None or end_date is None:
+            return requests.get(f'{self.base_url}/{lat},{lon}?unitGroup=metric&key={api_key}')
+        else:
+            start = start_date.strftime('%Y-%m-%d')
+            until = end_date.strftime('%Y-%m-%d')
+            return requests.get(f'{self.base_url}/{lat},{lon}/{start}/{until}?unitGroup=metric&key={api_key}')
```

Comment:
you can format values inside f-strings: 
```python
f"/{start_date:%Y-%m-%d}/{end_date:%Y-%m-%d}"
```

https://thetechplatform.medium.com/fstring-in-python-the-complete-guide-to-fstring-formatting-900676a768d5